pkg/build/buildapihelpers/helpers.go

```diff
@@ -50,3 +52,19 @@ func GetInputReference(strategy buildinternalapi.BuildStrategy) *coreinternalapi
 		return nil
 	}
 }
+
+// TODO(juanvallejo): change internal helper to "GetInputReferenceInternal" once https://github.com/openshift/origin/pull/20532 merges
+// GetInputReferenceV1 returns the From ObjectReference associated with the
+// BuildStrategy.
+func GetInputReferenceV1(strategy buildv1.BuildStrategy) *corev1.ObjectReference {
```

Comment:
Not like this, see how it's done in https://github.com/openshift/origin/pull/20532/files#diff-5cb0d8a77f515d00d8e768a169ded24c it'll be easier for both of us to have those in sync. Besides never create `*V1` versions, rather call the old one `*Internal` and let the external version re-use the name.

Reply:
You can probably just pick the commit that contains all those fixes from that PR.

Reply:
Will do - wanted to do it this way at first, but did not want to overlap changes in that PR. Seeing as how your PR will merge first anyway, I'll go ahead and apply your changes here